apps/meow/src/lib/storage.ts

```diff
@@ -372,9 +372,32 @@ export class MStore extends ZStore {
    * @param n number of messages (in reverse order) to list
    * @returns
    */
+  async getChannelFeed(channel: string, n:number):Promise<string[]> {
+   if (channel[0] !== `#`) { channel = '#' + channel; }
+   let channelFeed=[];
+    if (this.meowDbs.followingChannels.get(channel) === undefined) {
+      return [];
+    }
+
+    const channelDB = this.meowDbs.channels[channel];
+    if (channelDB === undefined) {
+      return [];
+    }
+
+    await channelDB.load(n); // load last "n" messages to memory
+    const messages = channelDB.iterator({
+      limit: n, reverse: true
+    }).collect();
+    for (const m of messages) {
+      const msg = m.payload.value as types.ZChainMessage;
+      const msgValue = await decode(msg.message,password)
+      channelFeed.push([msg,msgValue])
```

Comment:
Please return the msg as an object, in an array of messages. I think decryption should happen at the "printing"/"display" level. 

You should simply return `[ { ..m1 }, { ..m2 }, .. ,{ ..mN } ]` here

Reply:
Password won't be passed to every app that will use meow-app . like in zScreen messages should be received decoded